experimental/devices/mfrc522/mfrc522.go

```diff
@@ -130,65 +104,50 @@ func (r *Dev) String() string {
 //
 // It soft-stops the chip - PowerDown bit set, command IDLE
 func (r *Dev) Halt() error {
-	r.mu.Lock()
-	defer r.mu.Unlock()
-
-	if r.isWaiting {
-		select {
-		case <-r.stop:
-		default:
-		}
-
-		r.stop <- struct{}{}
-	}
-
-	return r.LowLevel.DevWrite(commands.CommandReg, 16)
-}
-
-// SetOperationTimeout updates the device timeout for card operations.
-//
-// Effectively that sets the maximum time the RFID device will wait for IRQ
-// from the proximity card detection.
-//
-//  timeout the duration to wait for IRQ strobe.
-func (r *Dev) SetOperationTimeout(timeout time.Duration) {
-	r.oMu.Lock()
-	defer r.oMu.Unlock()
-	r.operationTimeout = timeout
+	r.beforeCall()
+	defer r.afterCall()
+	return r.LowLevel.Halt()
 }
 
 // SetAntennaGain configures antenna signal strength.
 //
-//  gain - signal strength from 0 to 7.
+//  gain    signal strength from 0 to 7.
 func (r *Dev) SetAntennaGain(gain int) error {
+	r.beforeCall()
+	defer r.afterCall()
 	if gain < 0 || gain > 7 {
 		return wrapf("gain must be in [0..7] interval")
 	}
-
-	r.oMu.Lock()
-	defer r.oMu.Unlock()
-	r.antennaGain = gain
+	r.LowLevel.SetAntennaGain(gain)
 	return nil
 }
 
 // ReadCard reads the card sector/block.
```

Comment:
I'd prefer a smaller interface; I'm really not a fan of the Foo/FooTimed duplication. I'd recommend to:
- Remove the non timed version
- Rename the Timed one without the Timed suffix

What you can do is that when timeout is 0, use the default value. That will simplify the API.